bigtable/client/instance_admin.cc

```diff
@@ -108,6 +108,80 @@ google::bigtable::admin::v2::Instance InstanceAdmin::CreateInstanceImpl(
   return result;
 }
 
+std::future<google::bigtable::admin::v2::Cluster> InstanceAdmin::CreateCluster(
+    ClusterConfig instance_config, std::string const& instance_id,
+    std::string const& cluster_id) {
+  return std::async(std::launch::async, &InstanceAdmin::CreateClusterImpl, this,
+                    std::move(instance_config), instance_id, cluster_id);
+}
+
+google::bigtable::admin::v2::Cluster InstanceAdmin::CreateClusterImpl(
+    ClusterConfig const& cluster_config, std::string const& instance_id,
+    std::string const& cluster_id) {
+  // Copy the policies in effect for the operation.
+  auto rpc_policy = impl_.rpc_retry_policy_->clone();
+  auto backoff_policy = impl_.rpc_backoff_policy_->clone();
+
+  // Build the RPC request, try to minimize copying.
+  auto cluster = cluster_config.as_proto_move();
+  btproto::CreateClusterRequest request;
+  request.set_allocated_cluster(&cluster);
+  request.set_parent(project_name() + "/instances/" + instance_id);
+  request.set_cluster_id(cluster_id);
+
+  using ClientUtils =
+      bigtable::internal::noex::UnaryClientUtils<InstanceAdminClient>;
+
+  grpc::Status status;
+  auto response = ClientUtils::MakeCall(
+      *impl_.client_, *rpc_policy, *backoff_policy,
+      impl_.metadata_update_policy_, &InstanceAdminClient::CreateCluster,
+      request, "InstanceAdmin::CreateCluster", status, false);
+  if (not status.ok()) {
+    bigtable::internal::RaiseRpcError(status,
+                                      "unrecoverable error in MakeCall()");
+  }
+
+  google::bigtable::admin::v2::Cluster result;
+
+  while (not response.done()) {
```

Comment:
Unfortunately my implementation of `CreateInstance` had a bug which you reproduced here: if the longrunning operation is immediately satisfied (i.e. if `response.done() == true` the first time you enter this loop), the `result` variable is never initialized.  A recent merge has a new implementation without the bug, please fix.  We should consider refactoring too.